lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -1053,14 +1109,12 @@ static ValueDecl *deriveDecodable_init(TypeChecker &tc, Decl *parentDecl,
   // Func name: init(from: Decoder)
   DeclName name(C, C.Id_init, paramList);
 
-  auto *initDecl = new (C) ConstructorDecl(
-      name, SourceLoc(),
-      /*Failability=*/OTK_None,
-      /*FailabilityLoc=*/SourceLoc(),
-      /*Throws=*/true, /*ThrowsLoc=*/SourceLoc(), selfDecl, paramList,
-      /*GenericParams=*/nullptr, target);
+  auto *initDecl = new (C) ConstructorDecl(name, SourceLoc(), OTK_None,
+                                           SourceLoc(), /*Throws=*/true,
+                                           SourceLoc(), selfDecl, paramList,
+                                           /*GenericParams=*/nullptr, target);
   initDecl->setImplicit();
-  initDecl->setBodySynthesizer(deriveBodyDecodable_init);
+  initDecl->setBody(deriveBodyDecodable_init(tc, initDecl));
```

Comment:
If `super.init()` is synthesized, `DeclContext::lookupQualified` won't always be able to it to be synthesized unless we call `TypeChecker::addImplicitConstructors` from `TypeChecker::synthesizeMemberForLookup` (whereas `tc.lookupMember` does synthesize it if need be). The one danger to adding the `addImplicitConstructors` call is accidentally making `synthesizeMemberForLookup` recursive.

We also use the type checker to see if super conforms to `Decodable`, but we might be able to avoid that.